app/middleware/cmr-granule-locator.ts

```diff
@@ -134,6 +134,11 @@ async function cmrGranuleLocatorNew(
 
       operation.cmrHits += hits;
       operation.scrollIDs.push(scrollID);
+
+      const limitedMessage = getResultsLimitedMessage(operation);
+      if (limitedMessage) {
+        req.context.messages.push(getResultsLimitedMessage(operation));
```

Comment:
No need to call `getResultsLimitedMessage()` here since you have already saved the result of that call in `limitedMessage`.

Reply:
Makes me miss `(if-let`